src/app/product/[id]/page.tsx

```diff
@@ -0,0 +1,68 @@
+import { Metadata } from 'next'
+import Image from 'next/image'
+import React from 'react'
+import Counter from '../../../../components/Counter'
```

Comment:
I consider to put `Counter` component to separate package like`cart` because `Counter` relates to `cart` module.

Reply:
This component may be reused in the future, not only with the cart. It's common to store reusable components in the "components" folder.